rpc/rpc-lib/src/rpc-clnt.c

```diff
@@ -348,13 +350,15 @@ rpc_clnt_reconnect(void *conn_ptr)
     struct rpc_clnt *clnt = NULL;
     gf_boolean_t need_unref = _gf_false;
     gf_boolean_t canceled_unref = _gf_false;
+    glusterfs_ctx_t *ctx = NULL;
 
     conn = conn_ptr;
     clnt = conn->rpc_clnt;
+    ctx = clnt->ctx;
     pthread_mutex_lock(&conn->lock);
     {
         trans = conn->trans;
-        if (!trans)
+        if (!trans || ctx->cleanup_started)
```

Comment:
This is similar to the other case. Even though the check is done inside a mutex, it's not the same mutex used to set `cleanup_started`, so the race still exists.